client/java-spring-boot-autoconfigure/src/test/java/com/linecorp/centraldogma/client/spring/CentralDogmaInitializationTimeoutTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.client.spring;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.centraldogma.client.CentralDogma;
+import com.linecorp.centraldogma.client.spring.CentralDogmaClientAutoConfigurationWithDnsEndpointGroupConfiguratorTest.TestConfiguration;
+import com.linecorp.centraldogma.internal.api.v1.HttpApiV1Constants;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = TestConfiguration.class)
+@ActiveProfiles({ "local", "initTimeout" })
+class CentralDogmaInitializationTimeoutTest {
+
+    // TODO(ikhoon): Randomize the port number to avoid flakiness.
+    private static final Lock lock = new ReentrantLock();
+    private static final int TEST_SERVER_PORT = 56463;
```

Comment:
Why do we need this fixed port?

Reply:
Because `CentralDogma` client bean needs a pre-known hostname (`host:port`) to access a server.
https://github.com/line/centraldogma/pull/692/files#diff-4264d927674e9eb11e891891fce0d0305d917df8904caff14930b7df456b2a7f

Reply:
Ah, thanks for the explanation.
I'm a bit worried about the flakiness but we can fix it later if this causes the problem. 😉